### DISCUSSION TITLE
Why is the IP of the robot needed?

### DISCUSSION THREAD

**Discussion by Sonderstorch** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#379] [Q&A] [answered]
Hi there,

my Xiaomi robots are located in a separate guest WLAN for security reasons (well, I'm a coward). That guest WLAN uses a different IP network as the HomeAssistant server and devices in the Guest WLAN cannot see and cannot be seen from HomeAssistant's network.

But, the robots can access the internet and can controlled via the Xiaomi app using the Xiaomi cloud. My understanding is that this component does also collect the map from the cloud (it needs the credentials). Why is the local IP even required then?

**Comment by PiotrMachowski** [comment, marked as answer]
Which vacuum models do you have? Vacuums that use `xiaomi` API have to be accessible from HA for Map Extractor to work (it sends a command to a vacuum to retrieve the latest map)

**Reply by Sonderstorch** [reply]
The original Xiaomi Mi Vacuum and the Xiaomi Mi 1S; using the CN server.

**Reply by Sonderstorch** [reply]
Well, thinking about it: The Xiaomi App can retrieve and update the map without having the local IP. The update speed of the app would suffice for me, so may be an "cloud-only" option would be fine for an use case when the robots are not in the same IP subnet as HA.

Merry Christmas, Everyone.

**Reply by PiotrMachowski** [reply]
Yup, that's a possible improvement
